src/block-ref.ts

```diff
@@ -68,12 +63,17 @@ export function createBlockRefMiddleware({
 
     // perform child request
     log('Performing another request %o', childRequest);
-    const childRes: PendingJsonRpcResponse<Block> = await pify(
-      provider.sendAsync,
-    ).call(provider, childRequest);
     // copy child response onto original response
-    res.result = childRes.result;
-    res.error = childRes.error;
+    try {
+      const childResult = await provider.request<JsonRpcParams, Block>(
+        childRequest,
+      );
+      res.result = childResult;
+      res.error = undefined;
+    } catch (error: any) {
```

Comment:
Do you need the type annotation here?

``` suggestion
    } catch (error) {
```

If you do, then perhaps this should be `unknown`? `any` doesn't really mean "any" type — it's a shortcut for turning off type-checking altogether.

``` suggestion
    } catch (error: unknown) {
```